gatling-ss7/src/main/scala/io/snice/gatling/ss7/engine/Ss7Client.scala

```diff
@@ -99,43 +101,78 @@ class Ss7Client(mapStack: MAPStack,
     clientDialogSms.send()
   }
 
-  def sendCancelLocation(imsiStr: String): Unit = {
-//    logger.info(s"IMSI $imsiStr")
-//    //todo: this should be moved to the RequestBuilder where the params for the request are created
-//    val appCnt = MAPApplicationContext.getInstance(MAPApplicationContextName.locationCancellationContext, MAPApplicationContextVersion.version3)
-//    // First create Dialog
-//    val clientDialogMobility = mapProvider.getMAPServiceMobility.createNewDialog(appCnt, config.LOCAL_ADDRESS, null, config.REMOTE_ADDRESS, null)
-//    val imsi = mapParameterFactory.createIMSI(imsiStr)
-//    val lmsi = mapParameterFactory.createLMSI(config.LMSI_STR)
-//    val imsiWithLmsi = mapParameterFactory.createIMSIWithLMSI(imsi, lmsi)
-//
-//    clientDialogMobility.addCancelLocationRequest(imsi, imsiWithLmsi, config.CANCELLATION_TYPE, null, null, false, false, null, null, null)
-//    logger.info("addCancelLocationRequest OK")
-//    clientDialogMobility.send()
-//    logger.info("Sent CLR OK")
-
-    logger.info(s"IMSI $imsiStr")
-    //todo: this should be moved to the RequestBuilder where the params for the request are created
-    val appCnt = MAPApplicationContext.getInstance(MAPApplicationContextName.msPurgingContext, MAPApplicationContextVersion.version3)
+  def sendRequest(imsiStr: String, appCtx: MAPApplicationContext, callback: Status => Unit): Unit = {
     // First create Dialog
-    val clientDialogMobility = mapProvider.getMAPServiceMobility.createNewDialog(appCnt, config.LOCAL_ADDRESS, null, config.REMOTE_ADDRESS, null)
+    val clientDialogMobility = mapProvider.getMAPServiceMobility.createNewDialog(appCtx, config.LOCAL_ADDRESS, null, config.REMOTE_ADDRESS, null)
     val imsi = mapParameterFactory.createIMSI(imsiStr)
-    val sgsnNumber = new ISDNAddressStringImpl(AddressNature.international_number, NumberingPlan.ISDN, "22228")
 
-    clientDialogMobility.addPurgeMSRequest(imsi, null, sgsnNumber, null)
-    logger.info("addCancelLocationRequest OK")
+    // add callback
+    val transactionId = clientDialogMobility.getLocalDialogId
```

Comment:
Does the transactionId contain also length besides the ID itself? In TCAP the transaction ID might have a variable length, however I didn't see it being used with less than 4 bytes, which is the maximum.

Reply:
transactionID is just a `Long`